### PR TITLE
Input fixes

### PR DESCRIPTION
**Pull request by jusa**

### CODE REVIEW

File: src/common/droid-util.c [hunk 1]
```diff
@@ -1154,7 +1153,76 @@ static bool stream_config_fill(audio_devices_t devices,
...
+                    ddef->sampling_rates[i] != sample_spec->rate)
+                    continue;
+
+                pa_log_debug("Fill config: Override sample format, use %s.",
+                    pa_string_convert_num_to_str(CONV_STRING_FORMAT, hal_audio_format, &fmt) ? fmt : "<unknown>");
+                goto format_found;
+            }
+
+            /* Start from highest going to lowest */
+            for (i = 0; ddef->sampling_rates[i+1]; i++)
```
Comment (spiiroin) [inline src/common/droid-util.c]: (theoretical, but handled elsewhere?) empty array -> [0 + 1] goes out of bounds

File: src/droid/droid-source.c [hunk 2]
```diff
@@ -485,22 +486,22 @@ static pa_hook_result_t source_output_new_hook_callback(pa_core *c, pa_source_ou
...
+    u->source->channel_map = new_channel_map;
+    u->source->sample_spec = new_sample_spec;
+    pa_assert_se(pa_cvolume_remap(&u->source->reference_volume, &old_channel_map, &new_channel_map));
+    pa_assert_se(pa_cvolume_remap(&u->source->real_volume, &old_channel_map, &new_channel_map));
+    pa_assert_se(pa_cvolume_remap(&u->source->soft_volume, &old_channel_map, &new_channel_map));
+    if (reconfigured)
         pa_log_info("Source reconfigured.");
-    }
+    else
+        pa_log_info("Source reconfigure restored.");
```
Comment (spiiroin) [inline src/droid/droid-source.c]: nit: at least I would be baffled upon seeing "reconfigure restored"

File: src/common/droid-util.c [hunk 3]
```diff
@@ -1580,14 +1580,8 @@ void pa_droid_stream_unref(pa_droid_stream *s) {
...
         pa_log_debug("Destroy input stream %p", (void *) s);
-        set_active_input(s->module, NULL);
-        pa_mutex_lock(s->module->input_mutex);
         pa_idxset_remove_by_data(s->module->inputs, s, NULL);
-        if (s->input->stream) {
-            s->input->stream->common.standby(&s->input->stream->common);
-            s->module->device->close_input_stream(s->module->device, s->input->stream);
-        }
-        pa_mutex_unlock(s->module->input_mutex);
+        input_stream_close(s);
```
Comment (spiiroin) [inline src/common/droid-util.c]: should this too check that close input quirk?

Comment (jusa) [reply src/common/droid-util.c]: Nope, as this is happening when the whole stream is torn down.

File: src/droid/droid-source.c (hunk 2 above)
Comment (jusa) [reply src/droid/droid-source.c]: Updated logging to make a bit more sense. :)

File: src/common/droid-util.c (hunk 1 above)
Comment (jusa) [reply src/common/droid-util.c]: Updated and now empty array is checked in the beginnig of that block.

**Pull request merged**